jcl/src/openj9.traceformat/share/classes/com/ibm/jvm/traceformat/TraceFormat.java

```diff
@@ -355,21 +356,22 @@ public static void main(String[] args) throws Exception {
 				thread = current;
 				output.println(formatted.toString());
 			}
-			
+
 			/* print percentage */
 			if (context.getTotalRecords() != recordsProcessed) {
 				recordsProcessed = context.getTotalRecords();
 				float processedMbytes = ((float) (context.getTotalRecords() * context.getRecordSize())) / (1024 * 1024);
 				if (processedMbytes % 10 == 0) {
-					int percent = (int) ((processedMbytes * 100) / (totalBytes/(1024*1024)));
+					int percent = (int) ((processedMbytes * 100) / (totalBytes / (1024 * 1024)));
 					if (verbose.booleanValue()) {
-						end = System.currentTimeMillis();
-						float MbpsBlock = (float) 10.0 / (float) ((end - startBlock) / 1000.0);
-						float Mbps = (float) (processedMbytes) / (float) (((float) (end - start) / 1000.0));
-						startBlock = System.currentTimeMillis();
-						context.message(context, "Processed " + processedMbytes + "Mb ("+percent+"%), burst speed: "+MbpsBlock+"Mb/s, average: "+Mbps+"Mb/s");
+						end = System.nanoTime();
+						float MbpsBlock = (float) 10.0 / TimeUnit.NANOSECONDS.toSeconds(end - startBlock);
+						float Mbps = processedMbytes / TimeUnit.NANOSECONDS.toSeconds(end - start);
```

Comment:
Please correct this to avoid truncation.

Reply:
Updated.